products/billingbudget/api.yaml

```diff
@@ -0,0 +1,197 @@
+                      inclusive. If units is positive, nanos must be positive or
+                      zero. If units is zero, nanos can be positive, zero, or
+                      negative. If units is negative, nanos must be negative or
+                      zero. For example $-1.75 is represented as units=-1 and
+                      nanos=-750,000,000.
+              - !ruby/object:Api::Type::NestedObject
+                name: lastPeriodAmount
+                description: |
+                   Describes a budget amount targeted to last period's spend.
+                   At this time, the amount is automatically 100% of last
+                   period's spend; that is, there are no other options yet.
+                   Future configuration will be described here (for example,
+                   configuring a percentage of last period's spend).
+          - !ruby/object:Api::Type::Array
+            name: thresholdRules
+            required: true
+            description: |
+              Rules that trigger alerts (notifications of thresholds being
+              crossed) when spend exceeds the specified percentages of the
+              budget.
+            item_type: !ruby/object:Api::Type::NestedObject
+              properties:
+                - !ruby/object:Api::Type::Double
+                  name: thresholdPercent
+                  required: true
+                  description: |
+                    Send an alert when this threshold is exceeded. This is a
+                    1.0-based percentage, so 0.5 = 50%. Must be >= 0.
+                - !ruby/object:Api::Type::Enum
+                  name: spendBasis
+                  description: |
+                    The type of basis used to determine if spend has passed
+                    the threshold.
+                  default_value: :CURRENT_SPEND
+                  values:
+                    - :BASIS_UNSPECIFIED
+                    - :CURRENT_SPEND
+                    - :FORECASTED_SPEND
+          - !ruby/object:Api::Type::NestedObject
+            name: allUpdatesRule
+            description: |
+              Defines notifications that are sent on every update to the
+              billing account's spend, regardless of the thresholds defined
+              using threshold rules.
+            properties:
+              - !ruby/object:Api::Type::String
+                name: pubsubTopic
+                required: true
+                description: |
+                  The name of the Cloud Pub/Sub topic where budget related
+                  messages will be published, in the form
+                  projects/{project_id}/topics/{topic_id}. Updates are sent
+                  at regular intervals to the topic.
+              - !ruby/object:Api::Type::String
+                name: schemaVersion
+                required: true
+                description: |
+                  The schema version of the notification. Only "1.0" is
```

Comment:
Since only 1.0 is accepted here, it's slightly nicer UX to not allow users to set this field at all, and instead send it unconditionally (by adding an encoder)